devito/ir/iet/nodes.py

```diff
@@ -288,14 +288,12 @@ def is_tensor(self):
         return not self.is_scalar
 
     @property
-    def is_scalar_assign(self):
-        """True if a scalar, non-increment expression."""
-        return self.is_scalar and not self.is_Increment
-
-    @property
-    def is_tensor_assign(self):
-        """True if an array initialization"""
-        return self.is_tensor and isinstance(self.expr.rhs, ListInitializer)
+    def is_assign(self):
```

Comment:
is_definition

definition is the word in C that implies introduction of a symbol **and** allocation of memory (eg on the stack)

Reply:
otherwise, all expressions are actually assignments (well, not all, but most)